drivers/sensor/lm35/lm35.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT lm35
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#define LM35_GAIN       ADC_GAIN_1
+#define LM35_REF        ADC_REF_INTERNAL
+#define LM35_RESOLUTION 12
+
+LOG_MODULE_REGISTER(LM35, CONFIG_SENSOR_LOG_LEVEL);
+
+struct lm35_data {
+	struct adc_channel_cfg ch_cfg;
+	uint16_t raw;
+};
+
+struct lm35_config {
+	const struct device *adc;
+	uint8_t adc_channel;
+	struct adc_sequence adc_seq;
+};
+
+static int lm35_sample_fetch(const struct device *dev, enum sensor_channel chan)
```

Comment:
The sample fetch and get functions should only operate on supported channel types. For this kind of sensor, channels other than `SENSOR_CHAN_ALL` and `SENSOR_CHAN_AMBIENT_TEMP` should return `-ENOTSUP` in both fetch and get

Reply:
I see, I was not aware of this. Thanks, added this change